src/commands/format.rs

```diff
@@ -0,0 +1,118 @@
+//! Implementation of the format command.
+//! Implementation of the analyzer command.
+
+use std::fs;
+use std::io::Read;
+use std::num::NonZeroUsize;
+use std::path::Path;
+use std::path::PathBuf;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::anyhow;
+use anyhow::bail;
+use clap::Parser;
+use codespan_reporting::files::SimpleFile;
+use codespan_reporting::term::emit;
+use wdl::ast::Document;
+use wdl::ast::Node;
+use wdl::format::Formatter;
+use wdl::format::config::Builder;
+use wdl::format::config::Indent;
+use wdl::format::element::node::AstNodeFormatExt;
+
+use super::Mode;
+use crate::commands::get_display_config;
+
+/// Reads source from the given path.
+///
+/// If the path is simply `-`, the source is read from STDIN.
+fn read_source(path: &Path) -> Result<String> {
+    if path.as_os_str() == "-" {
+        let mut source = String::new();
+        std::io::stdin()
+            .read_to_string(&mut source)
+            .context("failed to read source from STDIN")?;
+        Ok(source)
+    } else {
+        Ok(fs::read_to_string(path).with_context(|| {
+            format!("failed to read source file `{path}`", path = path.display())
+        })?)
+    }
+}
+
+/// Arguments for the `analyzer` subcommand.
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+pub struct FormatArgs {
+    /// The path to the source WDL file (`-` for STDIN).
+    #[arg(value_name = "PATH")]
+    pub path: PathBuf,
+
+    /// Disables color output.
+    #[arg(long)]
+    pub no_color: bool,
+
+    /// The report mode.
+    #[arg(short = 'm', long, default_value_t, value_name = "MODE")]
+    pub report_mode: Mode,
+
+    /// Use tabs instead of spaces for indentation.
+    #[arg(long)]
+    pub with_tabs: bool,
+
+    /// The number of spaces that represents an indentation level.
+    #[arg(
+        long,
+        value_name = "SIZE",
+        default_value = "4",
+        conflicts_with = "with_tabs"
```

Comment:
I'm fine with this, but if we're drawing this line in the sand (that when using tabs, only one should be used for each "level"), we should probably update the `wdl-format` `Config` to match.

Reply:
We should also consider capping the number of spaces possible. Imagine someone supplied `MAX_INT` here. Nothing bad would happen but the file would be garbage. Maybe cap at `12`? `16`?

Reply:
Tangent, I've had a similar thought about how `max_line_width` could be set. There should be a minimum for that, but IDK what the magic number is. `60`? `40`? Why would you ever want a file narrower than that?

Reply:
It'd be weird to me to indent with multiple tabs, but can't hurt to be more flexible. I'll make the change along with capping it at a reasonable value.